### PR TITLE
feat(features): initial work on making cell nav conditional

### PR DESCRIPTION
**Pull request by JeffRiggle**
Initial work. This appears to be working to toggle between an active and inactive state. I think I will have to clean this up more but this is a first draft.

### CODE REVIEW

File: src/features/cellnav/js/cellnav.js [hunk 1]
```diff
@@ -641,132 +648,153 @@
...
-              };
-
-              uiGridCtrl.cellNav.broadcastFocus = function (rowCol, modifierDown, originEvt) {
-                modifierDown = !(modifierDown === undefined || !modifierDown);
-
-                rowCol = uiGridCtrl.cellNav.makeRowCol(rowCol);
 
-                var row = rowCol.row,
-                  col = rowCol.col;
+              function setupCellNav() {
```
Comment (zakeckert) [inline src/features/cellnav/js/cellnav.js]: I'm apparently really bad at indentation and stuff, but you wrapped all of this stuff in setupCellNav, right?

Comment (JeffRiggle) [reply src/features/cellnav/js/cellnav.js]: Basically yeah. Most of this stuff has just been moved into this function so we can toggle the behavior

**Comment by JeffRiggle** [comment]
Alright with these new revisions I think the feature is working as I would expect it to. Can everyone please take a read over this code before I start doing some refactoring. I think I need to move function definitions around a bit to make this code easier to read.

File: src/js/core/factories/GridApi.js [hunk 2]
```diff
@@ -249,16 +249,19 @@
             $rootScope.$emit.apply($rootScope, [eventId].concat(Array.prototype.slice.call(arguments)));
           };
 
+          var cleanup = [];
```
Comment (covertcj) [inline src/js/core/factories/GridApi.js]: This is basically the same thing as the destroySteps from above, right?  You may want to consider unifying their naming convention

File: src/features/cellnav/js/cellnav.js [hunk 3]
```diff
@@ -1064,119 +1131,160 @@
...
-              $elm.attr('aria-selected', true);
-              uiGridCellnavCtrl.setAriaActivedescendant($elm.attr('id'));
-              $scope.focused = true;
+          function tearDownFeature() {
+            if (dataChangeDereg) {
+              dataChangeDereg();
             }
+
+            destroySteps.forEach(function(destory) {
+              destory();
```
Comment (covertcj) [inline src/features/cellnav/js/cellnav.js]: Typo in destory

File: src/features/cellnav/js/cellnav.js [hunk 4]
```diff
@@ -1064,119 +1131,160 @@
...
+            // This event is fired for all cells.  If the cell matches, then focus is set
+            destroySteps.push($scope.$on(uiGridCellNavConstants.CELL_NAV_EVENT, refreshCellFocus));
 
-          function preventMouseDown(evt) {
-            //Prevents the foucus event from firing if the click event is already going to fire.
-            //If both events fire it will cause bouncing behavior.
-            evt.preventDefault();
-          }
+            // Refresh cell focus when a new row id added to the grid
+            dataChangeDereg = uiGridCtrl.grid.registerDataChangeCallback(function (grid) {
```
Comment (covertcj) [inline src/features/cellnav/js/cellnav.js]: If everything else just adds to destroySteps, why not do that here too?

File: src/js/core/factories/GridApi.js (hunk 2 above)
Comment (JeffRiggle) [reply src/js/core/factories/GridApi.js]: Yeah it's just scoped differently. I will start doing some real clean up an variable consistency type changes

**Pull request merged**